modules/oadp-timeouts.adoc

```diff
@@ -0,0 +1,138 @@
+[id="oadp-timeouts_{context}"]
+= OADP timeouts
+
+Extending a timeout allows complex or resource-intensive processes to complete successfully without premature termination, reducing the likelihood of errors, retries or failures. However, it is necessary to strike a balance when extending timeouts. Setting excessively long timeouts may hide underlying issues in the process. Therefore, careful consideration and monitoring are necessary to determine an appropriate timeout value that meets the needs of the process and the overall system performance.
+
+Below are various OADP timeouts, with instructions of how and when to implement these parameters:
+
+[id="Restic-timeout_{context}"]
+== Restic timeout
+
+.Kind
+DataProtectionApplication
+
+.Version
+oadp.openshift.io/v1alpha1
+
+.Field
+`timeout` <string>
+
+.Description
+The `timeout` defines the Restic timeout. The default value is 1h.
+
+.Implementation
+.. Use this parameter for Restic backups with large volumes.
+.. For Restic backups only. Use this parameter if backups are timing out with the following error:
++
+[source,terminal]
+----
+level=error msg="Error backing up item" backup=velero/monitoring error="timed out waiting for all PodVolumeBackups to complete"
+----
+
+[id="Velero-resource-timeout_{context}"]
+== Velereo resource timeout
+
+.Kind
+DataProtectionApplication
+
+.Version
+oadp.openshift.io/v1alpha1
+
+.Field
+`resourceTimeout` <string>
+
+.Description
+The `resourceTimeout` defines how long to wait for several Velero resources before timeout occurs, such as Velero CRD availability, `volumeSnapshot` deletion, and repo availability. The default is 10m.
+
+.Implementation
+.. Use this timeout for backups with very large volumes . This parameter is used as a timeout value when Velero tries to clean-up or delete the Container Storage Interface (CSI) snapshots, before marking the backup as complete.
+... A sub-task of this cleanup tries to patch VSC and this timeout can be used for that task.
++
+.. Use this timeout to create or ensure a backup repository is ready for filesystem based backups for Restic or Kopia.
+.. Use this timeout to check if the Velero CRD is available in the cluster before restoring the custom resource (CR) or resource from the backup.
+
+[id="Velero-default-item-operation-timeout_{context}"]
+== Velereo default item operation timeout
+
+.Kind
+DataProtectionApplication
+
+.Version
+oadp.openshift.io/v1alpha1
+
+.Field
+`defaultItemOperationTimeout` <string>
+
+.Description
+How long to wait on asynchronous `BackupItemActions` and `RestoreItemActions` to complete before timing out. The default value is 1h.
+
+.Implementation
+.. Use this timeout only with Data Mover.
+.. Use this timeout to specify the amount of time a particular backup should wait for the Asynchronous actions to complete. In the context of OADP features, this value is used for the Asynchronous actions involved in the Container Storage Interface (CSI) Data Mover feature.
```

Comment:
This timeout also apply for restore, please update.

Reply:
I added.
"the amount of time a particular backup or restore should wait for the Asynchronous..."